server/item/enchantment/thorns.go

```diff
@@ -0,0 +1,24 @@
+package enchantment
+
+import (
+	"github.com/df-mc/dragonfly/server/item"
+)
+
+// Thorns is an enchantment that inflicts damage on attackers.
+type Thorns struct{}
```

Comment:
Make sure to update this enchantment to include the new functions introduced with anvils.